hooks/useFetch.js

```diff
@@ -73,27 +76,20 @@ const getURL = (url) =>
     materialize(),
     filter((i) => !isComplete(i)),
   )
+}
 
 const cache = {}
 
 const fetchData = (url) => {
   if (url in cache) return cache[url]
 
-  const loading = new BehaviorSubject(false)
-
-  const data = new BehaviorSubject(url).pipe(
-    tap(() => loading.next({ url: url, status: true })),
-    switchMap(getURL),
-    tap(() => loading.next({ url: url, status: false })),
-    shareReplay(1),
-  )
-
-  const refresh = combineLatest(of(url), interval(1000)).pipe(
-    pluck(0),
-    tap(data),
+  const refresher = timer(0, 1000).pipe(
+    switchMap(() => getURL(url)),
+    publish(),
+    refCount(),
```

Comment:
Why do publish/refCount instead of just `share` (or just using a subject)? From the docs:

> Note that using the share operator is exactly the same as using the publish operator (making the observable hot) and the refCount operator in a sequence.

Reply:
Yes, you could use `share`. However, depending on the use case you might want to use `publishBehavior` or `publishReplay` to get different characteristics some of which have their own `shareReplay` etc. Just got used to using them in conjunction since it's easier to play around with them individually. - Once you've got the right behavior you could condense a sequence of operator with combinator operatoers.